content/shmem_broadcast.tex

```diff
@@ -72,12 +72,15 @@
     needed to ensure this): The \VAR{pSync} array on all \ac{PE}s in the
     \activeset{} is not still in use from a prior call to a broadcast routine.  The
     \dest{} array on all \ac{PE}s in the \activeset{} is ready to accept the
-    broadcast data.
+    broadcast data. \newtext{The data in the \source{} array on the root \ac{PE} is ready to be broadcasted.}
+    
+    \newtext{The \source{} and \dest{} arrays may be the same array, but they may not be overlapping arrays. If they 
+    are different, the source array on the non-root \acp{PE} and \dest{} array on the root \ac{PE} is not used by the broadcast routine.}
```

Comment:
The dest array in the root PE shouldn't be made visible to the application when the broadcast is in progress, because it is being used by the HW operations. But, this specific change guarantees that both the dest array in root PE and source array in the non-root PEs are application-visible.